### PR TITLE
skip records with arrival time older than ArchivingCutoffHighWatermark

### PR DESCRIPTION
**Pull request by jshencode**

### CODE REVIEW

File: memstore/ingestion.go [hunk 1]
```diff
@@ -206,6 +206,9 @@ func (shard *TableShard) insertPrimaryKeys(primaryKeyColumns []int, eventTimeCol
 				return nil, nil, nil, utils.StackError(err, "Event time for row %d is null", row)
 			}
 			primaryKeyEventTime = upsertBatch.ArrivalTime
+			if primaryKeyEventTime < shard.LiveStore.ArchivingCutoffHighWatermark {
```
Comment (voyager-dw) [inline memstore/ingestion.go]: it seems for normal event, we put into backfill queue, is there any other purge time that we even dont need put into backfill queue, or handled by backfill?

if this is for yesterday's issue, we need to fix in redolog recovery too

Comment (jshencode) [reply memstore/ingestion.go]: 1.  in total we have two cases:
    event with valid **event time**, we place into backfill queue once **event time** is older than archiving cutoff
    event with invalid **event time**, we ignore the event when **arrival time** is older than archiving cutoff 
2. this code is shared for recovery and ingestion

Comment (voyager-dw) [reply memstore/ingestion.go]: I think yesterday's case the event time is valid, but older than the cutoff time when recover, no?

Comment (shz117) [reply memstore/ingestion.go]: I think if it hits this branch the records `!isEventTimeValid`

Comment (voyager-dw) [reply memstore/ingestion.go]: i see, both ingestion and recovery using this function

Comment (lucafuji) [inline memstore/ingestion.go]: add comments saying that those missing event time records will not go to backfill queue if they are late

Comment (lucafuji) [inline memstore/ingestion.go]: they can be combined into the previous line
```
if  primaryKeyEventTime = upsertBatch.ArrivalTime; primaryKeyEventTime < shard.LiveStore.ArchivingCutoffHighWatermark {
```

Comment (jshencode) [reply memstore/ingestion.go]: done

Comment (jshencode) [reply memstore/ingestion.go]: done

**Pull request merged**